### DISCUSSION TITLE
[ttx] adding an undroppable table to protect font copyright

### DISCUSSION THREAD

**Discussion by aminabedi68** [fonttools/fonttools#3263] [Ideas]
Hi
everyone could use ttx to modify a commercial font name table entries and this cause some concern about copyright protection. i suggest add an **undroppable** (and uneditable)table that generates automatically when copyright info (and/or most of important name entries) changes during a process and save the original data in that table and shown on the top of all tables. this could be a valid evidence.

i don't know it's possible at all or not. maybe a backup of original file hash data in new file attributes would also helps or any kind of smarter ways, and i'm not sure that hex editors could erase all these kind of protections or not.

**Comment by justvanrossum** [comment]
- You can’t distinguish between “illegal” edits and legitimate edits
- There are other edit tools besides TTX
- Many font editors will rebuild a font from scratch
- Should a tool like TTX be responsible for policing copyright violations? (No.)
- Etc.

**Reply by aminabedi68** [reply]
Thank you @justvanrossum 

- nobody wants to distinguish between legal/illegal edits, this is impossible and insulting, the idea is to preserve first name/copyright data somehow(before ttx) to probable feature looks and compares. so the logic isn't to judge the user act.

- you're right, The other editors has the same issue. so this is not gonna work with only ttx, but also most of them(if I'm not mistaken) are paid products or not as famous/available as ttx.

- font editors can't open their own generated variable or color fonts correctly, and in many cases have many bugs in simple fonts, but ttx could open and edit name table very clean!

- we are not some business man, we love our works but the concerns also are real.

What is your opinion? Is it technically possible? please share your thoughts.

**Comment by behdad** [comment]
It's not technically possible, no. If we do, someone will just fork fonttools and remove it.

That's the reality of Open Source software.

**Reply by aminabedi68** [reply]
thanks @behdad
absolutely true.
